.github/workflows/ci-build.yaml

```diff
@@ -276,20 +258,16 @@ jobs:
           echo "  user:" >> $KUBECONFIG
           echo "    token: xxxxxx" >> $KUBECONFIG
           until kubectl cluster-info ; do sleep 10s ; done
-      - name: Download argoexec image
-        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
-        with:
-          name: argoexec
-          path: /tmp
-      - name: Load argoexec image
-        run: docker load < /tmp/argoexec_image.tar
-      - name: Download argocli image
+      - name: Download images
         uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
         with:
-          name: argocli
+          pattern: '*_image.tar'
```

Comment:
> It seems like ["download all" is available](https://github.com/actions/download-artifact/tree/v3.0.2?tab=readme-ov-file#download-all-artifacts) if you don't specify `name` and use a directory.

Annnd that appears to be what we're using here since `pattern` is invalid in `actions/download-artifact@v3`.

This is getting a warning [in CI](https://github.com/argoproj/argo-workflows/actions/runs/9249013492) for every E2E test in the matrix:
```
[E2E Tests (test-executor, minimal)](https://github.com/argoproj/argo-workflows/actions/runs/9249013492/job/25442227788#step:8:1)
Unexpected input(s) 'pattern', valid inputs are ['name', 'path']
```
![Screenshot 2024-05-27 at 1 44 33 AM](https://github.com/argoproj/argo-workflows/assets/4970083/634f2c23-62e5-4791-997e-82ffba1f5f5d)